Tella/Components/NavigationHeaderView.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  NavigationHeaderView.swift
+//  Tella
+//
+//  Created by gus valbuena on 4/10/24.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import SwiftUI
+
+enum NavigationType {
+    case save
+    case reload
+
+    var imageName: String {
+        switch self {
+        case .save: return "report.select-files"
+        case .reload: return "arrow.clockwise"
+        }
+    }
+}
+
+struct NavigationHeaderView: View {
+    var backButtonAction : () -> Void
+    var reloadAction: () -> Void
```

Comment:
I think it's not in all cases is `reloadAction` and there is a shared `backButtonAction` ( `presentationMode.wrappedValue.dismiss()` ) which we can include it in the component, also we can add `.none` type if there is no right button.
What do you think if we change it to this ? : 


```
enum NavigationType {
    case none
    case validate
    case save
    case reload
    case delete
    
    var imageName: String {
        switch self {
        case .validate: return "report.select-files"
        case .save: return "reports.save"
        case .reload: return "arrow.clockwise"
        case .delete: return "report.delete-outbox"
        case .none: return ""
        }
    }
}

```

```
struct NavigationHeaderView: View {
    @Environment(\.presentationMode) var presentationMode: Binding<PresentationMode>
    
    var title: String = ""
    var type: NavigationType = .none
    var backButtonAction : (() -> Void)? = nil
    var rightButtonAction: (() -> Void)? = nil
    
    var body: some View {
        HStack(spacing: 0) {
            backButton
            headerTitleView
            Spacer()
            if(type != .none) {
                rightButton
            }
        }.frame(height: 56)
            .padding(.horizontal, 16)
    }
    
    private var backButton: some View {
        Button {
            presentationMode.wrappedValue.dismiss()
            backButtonAction?()
        } label: {
            Image("back")
                .padding(EdgeInsets(top: 10, leading: 0, bottom: 5, trailing: 12))
        }
    }
    
    private var headerTitleView: some View {
        Text(title)
            .font(.custom(Styles.Fonts.semiBoldFontName, size: 18))
            .foregroundColor(Color.white)
    }
    
    private var rightButton: some View {
        Button(action: { rightButtonAction?() }) {
            Image(type.imageName)
                .resizable()
                .frame(width: 24, height: 24)
        }
    }
}
```

and for implementation in :

```
 NavigationHeaderView(title:prompt.question,
                       type: .validate,
                       rightButtonAction:{presentationMode.wrappedValue.dismiss()})

```